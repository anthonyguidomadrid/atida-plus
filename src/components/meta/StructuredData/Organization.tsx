import { Organization as OrganizationType } from 'schema-dts'
import { jsonLdScriptProps } from 'react-schemaorg'
import { useRouter } from 'next/router'
import Head from 'next/head'
import { useSelector } from 'react-redux'
import getConfig from 'next/config'
import { selectOrganization } from '~domains/page'

export const Organization = () => {
  const { publicRuntimeConfig } = getConfig()
  const host = publicRuntimeConfig.host
  const organization = useSelector(selectOrganization)
  const { locale } = useRouter()

  return organization ? (
    <Head>
      <script
        key="organization-structured-data"
        type="application/ld+json"
        {...jsonLdScriptProps<OrganizationType>({
          '@context': 'https://schema.org',
          '@type': 'Organization',
          name: organization.name,
          description: organization.description,
          telephone: organization.telephone,
          email: organization.email,
          sameAs: organization.sameAs,
          image: organization.imageUrl,
          url: `${host}/${locale}`,
          ...(organization.address && {
            address: {
              '@type': 'PostalAddress',
              ...organization.address
            }
          })
        })}
      />
    </Head>
  ) : null
}
