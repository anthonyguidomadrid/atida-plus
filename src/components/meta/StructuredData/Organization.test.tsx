import { renderWithStore } from '~test-helpers'
import { Organization } from './Organization'

describe(Organization, () => {
  it('renders script tag with json-ld data', () => {
    const { container } = renderWithStore(<Organization />, {
      initialState: {
        server: {
          page: {
            common: {
              isLoading: false,
              wasSuccess: true,
              wasError: false,
              organization: {
                name: 'Atida',
                description: 'Farmácia Online de Confiança',
                telephone: '+34 967115501',
                email: 'https://support.atida.com/pt-PT/support/home',
                sameAs: [
                  'https://twitter.com/atida_pt,',
                  'https://www.facebook.com/AtidaPT/',
                  'https://www.instagram.com/atida_pt/'
                ],
                imageUrl:
                  'https://www.atida.com/pt-pt/blog/wp-content/uploads/2021/08/atida-mifarma-logo.png',
                id: 'atida-pt',
                address: {
                  streetAddress: '10 Downing Street',
                  addressLocality: 'London',
                  addressRegion: 'London',
                  postalCode: 'SW1A 2AA',
                  addressCountry: 'UK'
                }
              }
            }
          }
        }
      }
    })
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders nothing when organization data is not found', () => {
    const { container } = renderWithStore(<Organization />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
