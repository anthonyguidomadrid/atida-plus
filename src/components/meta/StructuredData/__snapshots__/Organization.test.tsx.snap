// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization renders nothing when organization data is not found 1`] = `null`;

exports[`Organization renders script tag with json-ld data 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"Organization","name":"Atida","description":"Farmácia Online de Confiança","telephone":"+34 967115501","email":"https://support.atida.com/pt-PT/support/home","sameAs":["https://twitter.com/atida_pt,","https://www.facebook.com/AtidaPT/","https://www.instagram.com/atida_pt/"],"image":"https://www.atida.com/pt-pt/blog/wp-content/uploads/2021/08/atida-mifarma-logo.png","url":"somehost.com/en-gb","address":{"@type":"PostalAddress","streetAddress":"10 Downing Street","addressLocality":"London","addressRegion":"London","postalCode":"SW1A 2AA","addressCountry":"UK"}}
</script>
`;
